### DISCUSSION TITLE
The Future

### DISCUSSION THREAD

**Discussion by tmedwards** [tmedwards/sugarcube-2#315] [General]
I've been thinking about v3 and I've come to the conclusion that if I wait until every improvement that I want to make is ready that it'll never be done, at least in a reasonable time frame.  I think I've been reluctant to advance the version number and really cut loose with the changes, for reasons that are probably historic.

What I think I've decided is to be a little freer with the versioning.  Now, I don't mean jumping on the version roller coaster, like browser manufacturer's for instance, just that I'll be less adverse to bumping the version number.

# Roadmap

What does this mean going forward?  Well, it means this:

## v1

End of life.

## v2

It's time to enter maintenece mode and end new feature support.

v2 has had it's day, but it's time to put it out to pasture.

## v3

Is not going to contain *all* the features that were planned.  Likely this means that it is going to be a cleaned up and almost evergreen v2.  Evergreen here meaning supporting close to web standards, with close meaning within six months to year of full support for features to be supported.  No longer will it instead be pinned to a particular browser version, as with v2.

The feature that is probably getting lost is the parser refresh, which is likely to become a v4 feature.

## v4

Will likely contain whatever features don't make it into v3.  Most notably the parser refresh.

**Comment by ChapelR** [comment]
I think moving to "evergreen" support is a great idea. 

I feel like 3 will become almost a transition version, and may turn into just being an alpha for 4. That might not be a bad thing, though. Alsop I guess it depends on how distant 4 will be from 3. I think putting 2 into maintenance is a good idea though, clearly a lot of development energy is still going into it, and you have to cut it off at some point.
